base/errorshow.jl

```diff
@@ -888,6 +888,18 @@ end
 
 Experimental.register_error_hint(noncallable_number_hint_handler, MethodError)
 
+# handler for displaying a hint in case the user tries to setindex
+# the instance of a number (probably attempting to use wrong indexing)
+# eg: a = [1 2; 3 4]; a[1][2] = 5
+function nonsetable_number_hint_handler(io, ex, arg_types, kwargs)
+    if ex.f == setindex! && arg_types[1] <: Number
```

Comment:
`kwargs` is a NT, so it must be marked `@nospecialize`

```suggestion
    if ex.f === setindex! && arg_types[1] <: Number
```

Reply:
NT = named tuple? And that's because there can be many combinations of such and you don't want the compiler to recompile for each one, yes?

Reply:
Yes, but now it is an Array{Any}, so this is no longer necessary